README.md

# PakIndicatorsApp

[![Lifecycle:
experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)

The goal of PakIndicatorsApp is to provide a scaffolding for the
schematic and efficient management and visualization of the district
level indicators of Pakistan, for multiple rounds of the household
surveys. It covers a variety of socio-economic (SDG related) indicators
ranging from poverty incidence, labor market outcomes, health and
nutrition, stunting and wasting, food security and so on. We will be
updating the app based on the data availability and revisions/feedback.

## Data Coverage

Currently, the portal covers HIES, PLSM, MICS and NNS surveys. (various
rounds from 2004 to 2020)

## Usage

The app can specifically be used to explore spatial disparities in
development outcomes at district (adm2) level in the country.

## Installation

You can install the development version of PakIndicatorsApp like so:

``` r
# To install the application 
remotes::install_github("szhaider/PakIndicatorsApp")

# To launch the app locally
library(PakIndicatorsApp)
PakIndicatorsApp::run_app()

#To run directly from Github
library(shiny)
shiny::runGitHub("PakIndicatorsApp", "szhaider")
```

## Example

These are some basic details about the data used in the application:

## Doamins covered

    #>  [1] "Household Welfare"           "Water and Sanitation"       
    #>  [3] "Affordable and Clean Energy" "Asset Ownership"            
    #>  [5] "Health and Wellbeing"        "Transportation"             
    #>  [7] "Household Charactersitics"   "Dwelling Charactersitics"   
    #>  [9] "Labor Market"                "Quality Education"          
    #> [11] "Demographics"                "Childhood Development"      
    #> [13] "Gender Equality"             "Food Insecurity"            
    #> [15] "Malnutrition"                "Social Protection"          
    #> [17] "Utilities"                   "Vaccination"                
    #> [19] "Maternal Newborn Care"

## Years Covered

    #>  [1] 2004 2006 2008 2010 2012 2014 2016 2017 2018 2019 2020

## Surveys over the years

    #>  [1] "2004-HIES/PSLM"                   "2004-PSLM"                       
    #>  [3] "2004-MICS"                        "2004-Census"                     
    #>  [5] "2006-HIES/PSLM"                   "2006-PSLM"                       
    #>  [7] "2006-MICS"                        "2006-Census"                     
    #>  [9] "2008-HIES/PSLM"                   "2008-PSLM"                       
    #> [11] "2008-MICS"                        "2008-Census"                     
    #> [13] "2010-HIES/PSLM"                   "2010-PSLM"                       
    #> [15] "2010-MICS : Punjab & Balochistan" "2010-Census"                     
    #> [17] "2012-HIES/PSLM"                   "2012-PSLM"                       
    #> [19] "2012-MICS"                        "2012-Census"                     
    #> [21] "2014-HIES/PSLM"                   "2014-PSLM"                       
    #> [23] "2014-MICS : Punjab & Sindh"       "2014-Census"                     
    #> [25] "2016-HIES/PSLM"                   "2016-PSLM"                       
    #> [27] "2016-MICS : KP"                   "2016-Census"                     
    #> [29] "2017-HIES/PSLM"                   "2017-PSLM"                       
    #> [31] "2017-MICS"                        "2018-Census"                     
    #> [33] "2018-PSLM"                        "2018-MICS : Punjab"              
    #> [35] "2017-Census"                      "2018-HIES/PSLM"                  
    #> [37] "2018-NNS"                         "2019-Census"                     
    #> [39] "2019-PSLM"                        "2019-MICS : KP & Sindh"          
    #> [41] "2020-MICS : Balochistan"

## Indicators covered

    #>   [1] "Poverty Rate (%)"                                                                                                                      
    #>   [2] "National poverty rank (N)"                                                                                                             
    #>   [3] "Provincial poverty rank (N)"                                                                                                           
    #>   [4] "Number of poor (1,000s)"                                                                                                               
    #>   [5] "Access to improved drinking water (% of population)"                                                                                   
    #>   [6] "Access to piped water (% of population)"                                                                                               
    #>   [7] "Access to improved drinking water, excl. piped water (% of population)"                                                                
    #>   [8] "Access to improved water within 30 minutes round trip"                                                                                 
    #>   [9] "Access to improved toilet facilities (% of population)"                                                                                
    #>  [10] "Access to flush toilet connected to sewer (% of population)"                                                                           
    #>  [11] "Access to flush toilet connected to septic tank (% of population)"                                                                     
    #>  [12] "Open defecation (% of population)"                                                                                                     
    #>  [13] "Access to electricity (% of population)"                                                                                               
    #>  [14] "Reliance on clean fuels for cooking (% of population)"                                                                                 
    #>  [15] "Households' mobile phone ownership (% of population)"                                                                                  
    #>  [16] "Accessibility, health (% of population within 15 minutes to clinic/hospital)"                                                          
    #>  [17] "Accessibility, transportation (% of population within 15 minutes to public transportation)"                                            
    #>  [18] "Household size"                                                                                                                        
    #>  [19] "Household dependency ratio"                                                                                                            
    #>  [20] "Household child dependency ratio"                                                                                                      
    #>  [21] "Household senior dependency ratio"                                                                                                     
    #>  [22] "Occupancy, present occupancy status-owner (% of population)"                                                                           
    #>  [23] "Occupancy, present occupancy status-rented (% of population)"                                                                          
    #>  [24] "Occupancy, present occupancy status-other (% of population)"                                                                           
    #>  [25] "Improved roof material (% of population)"                                                                                              
    #>  [26] "Improved wall material (% of population)"                                                                                              
    #>  [27] "Improved roof and wall material (% of population)"                                                                                     
    #>  [28] "Household members per room"                                                                                                            
    #>  [29] "Overcrowding (% of population)"                                                                                                        
    #>  [30] "Labor force participation rate (% of working age population, 15-64 years old)"                                                         
    #>  [31] "Employment in agriculture (% of total employment)"                                                                                     
    #>  [32] "Employment in industry (% of total employment)"                                                                                        
    #>  [33] "Employment services (% of total employment)"                                                                                           
    #>  [34] "Wage and salaried employment (% of total employment)"                                                                                  
    #>  [35] "Self-employment, non-agriculture (% of total employment)"                                                                              
    #>  [36] "Self-employment, agriculture (% of total employment)"                                                                                  
    #>  [37] "Unpaid employment (% of total employment)"                                                                                             
    #>  [38] "Child labor (% of children aged 10-17)"                                                                                                
    #>  [39] "Labor force participation rate, male (% of male working age population, 15-64 years old)"                                              
    #>  [40] "Employment in agriculture, male (% of male employment)"                                                                                
    #>  [41] "Employment in industry, male (% of male employment)"                                                                                   
    #>  [42] "Employment services, male (% of male employment)"                                                                                      
    #>  [43] "Wage and salaried employment, male (% of male employment)"                                                                             
    #>  [44] "Self-employment, non-agriculture, male (% of male employment)"                                                                         
    #>  [45] "Self-employment, agriculture, male (% of male employment)"                                                                             
    #>  [46] "Unpaid employment, male (% of male employment)"                                                                                        
    #>  [47] "Labor force participation rate, female (% of female working age population, 15-64 years old)"                                          
    #>  [48] "Employment in agriculture, female (% of female employment)"                                                                            
    #>  [49] "Employment in industry, female (% of female employment)"                                                                               
    #>  [50] "Employment services, female (% of female employment)"                                                                                  
    #>  [51] "Wage and salaried employment, female (% of female employment)"                                                                         
    #>  [52] "Self-employment, non-agriculture, female (% of female employment)"                                                                     
    #>  [53] "Self-employment, agriculture, female (% of female employment)"                                                                         
    #>  [54] "Unpaid employment, female (% of female employment)"                                                                                    
    #>  [55] "Gross school enrollment, primary school, 6-10 years old (% gross)"                                                                     
    #>  [56] "Gross school enrollment, middle school, 11-13 years old (% gross)"                                                                     
    #>  [57] "Net school enrollment, primary school, 6-10 years old (% net)"                                                                         
    #>  [58] "Net school enrollment, middle school, 11-13 years old (% net)"                                                                         
    #>  [59] "Gross school enrollment, primary school, 6-10 years old, male (% gross male)"                                                          
    #>  [60] "Gross school enrollment, middle school, 11-13 years old, male (% gross male)"                                                          
    #>  [61] "Net school enrollment, primary school, 6-10 years old, male (% net male)"                                                              
    #>  [62] "Net school enrollment, middle school, 11-13 years old, male (% net male)"                                                              
    #>  [63] "Gross school enrollment, primary school, 6-10 years old, female (% gross female)"                                                      
    #>  [64] "Gross school enrollment, middle school, 11-13 years old, female (% gross female)"                                                      
    #>  [65] "Net school enrollment, primary school, 6-10 years old, female (% net female)"                                                          
    #>  [66] "Net school enrollment, middle school, 11-13 years old, female (% net female)"                                                          
    #>  [67] "Youth literacy, 15-24 years old (% of population aged 15-24)"                                                                          
    #>  [68] "Youth literacy, 15-24 years old, female (% of women aged 15-24)"                                                                       
    #>  [69] "Youth literacy, 15-24 years old male (% of men aged 15-24)"                                                                            
    #>  [70] "Adult literacy, 25 or more years old (% of population aged 25 or more)"                                                                
    #>  [71] "Adult literacy, 25 or more years old, female (% of women aged 25 or more)"                                                             
    #>  [72] "Adult literacy, 25 or more years old, male (% of men aged 25 or more)"                                                                 
    #>  [73] "Primary completion rate, 14-17 years old (% of population aged 14-17)"                                                                 
    #>  [74] "Youth literacy gender parity index, 15-24 years old"                                                                                   
    #>  [75] "Adult literacy gender parity index, 25 or more years old"                                                                              
    #>  [76] "Primary completion gender parity index, 14-17 years old"                                                                               
    #>  [77] "Gross enrolment rate in primary gender parity index, 6-10 years old"                                                                   
    #>  [78] "Gross enrolment rate in middle school gender parity index, 11-13 years old"                                                            
    #>  [79] "Net enrolment rate in primary gender parity index, 6-10 years old"                                                                     
    #>  [80] "Net enrolment rate in middle school gender parity index, 11-13 years old"                                                              
    #>  [81] "Diarrhea incidence (% of children under 5)"                                                                                            
    #>  [82] "Diarrhea Treatment with Oral Rehydration Salts (% of children under 5 who had diarrhea during a 30-day recall period)"                 
    #>  [83] "Pregnant women receiving antenatal care (% of women  aged 15-49 who gave birth in last 3 years)"                                       
    #>  [84] "Assisted birth at healthcare facility (% of women aged 15-49 who gave birth in last 3 years)"                                          
    #>  [85] "Immunization, fully immunized (% children 12-24 months)"                                                                               
    #>  [86] "Immunization, polio three doses (% children 12-24 months)"                                                                             
    #>  [87] "Immunization: diphtheria, pertussis, and tetanus - three doses (% children 12-24 months)"                                              
    #>  [88] "Urban share of population (% of population)"                                                                                           
    #>  [89] "Access to improved drinking water - MICS (% of population)"                                                                            
    #>  [90] "Access to piped water - MICS (% of population)"                                                                                        
    #>  [91] "Access to motorized pump (% of population)"                                                                                            
    #>  [92] "Access to hand pump (% of population)"                                                                                                 
    #>  [93] "Access to improved toilet facilities - MICS (% of population)"                                                                         
    #>  [94] "Access to flush toilet connected to sewer - MICS (% of population)"                                                                    
    #>  [95] "Access to flush toilet connected to septic tank - MICS (% of population)"                                                              
    #>  [96] "Open defecation - MICS (% of population)"                                                                                              
    #>  [97] "Women's marriage before age 15 (% of women aged 20-24)"                                                                                
    #>  [98] "Women's marriage before age 18 (% of women aged 20-24)"                                                                                
    #>  [99] "Average age at first marriage of women (married woman aged 15 to 49)"                                                                  
    #> [100] "Child registration (% of children aged under 5)"                                                                                       
    #> [101] "Prevalence of stunting, height for age (% of children under 5)"                                                                        
    #> [102] "Prevalence of wasting, weight for height (% of children under 5)"                                                                      
    #> [103] "Prevalence of underweight, weight for age (% of children under 5)"                                                                     
    #> [104] "Diarrhea incidence (% of children under 5 who had diarrhea during a 2-week recall period)"                                             
    #> [105] "Early initiation of breastfeeding (% of recent mothers)"                                                                               
    #> [106] "Exclusive breastfeeding (% of children under 6 months)"                                                                                
    #> [107] "Children given plain water (% of children under 6 months)"                                                                             
    #> [108] "Children given milk (% of children under 6 months)"                                                                                    
    #> [109] "Children given baby formula (% of children under 6 months)"                                                                            
    #> [110] "Children left unattended (% of children under 5)"                                                                                      
    #> [111] "Toilet use by children (% of children 12-35 months old)"                                                                               
    #> [112] "Diarrhea treatment- less than regular liquid (% of children under 5 who had diarrhea episode in the two weeks prior to the survey)"    
    #> [113] "Diarrhea treatment- less than regular food (% of children under 5 who had a diarrhea episode in the two weeks prior to the survey)"    
    #> [114] "Diarrhea Treatment with Oral Rehydration Salts (% of children under 5 who had a diarrhea episode in the two weeks prior to the survey)"
    #> [115] "Pregnant women receiving prenatal care (% of ever-married women aged 15-49 with a live birth in the 2 years before the survey)"        
    #> [116] "Assisted birth (% of ever-married women aged 15-49 with a live birth in the 2 years before the survey)"                                
    #> [117] "Average age at first birth of women"                                                                                                   
    #> [118] "Immunization, BCG for Tuberculosis (% of children under 3)"                                                                            
    #> [119] "Immunization, Polio at birth (% of children under 3)"                                                                                  
    #> [120] "Immunization, Polio-first dose (% of children under 3)"                                                                                
    #> [121] "Immunization, Polio-second dose (% of children under 3)"                                                                               
    #> [122] "Immunization, Polio-third dose (% of children under 3)"                                                                                
    #> [123] "Immunization, Penta-1 (% of children under 3)"                                                                                         

NEWS.md
# PakIndicatorsApp 0.0.0.9000

* Added a `NEWS.md` file to track changes to the package.


R/mod_main_maps.R
#' main_maps UI Function
#'
#' @description Main Maps on the landing page.
#'
#' @param id,input,output,session Internal parameters for {shiny}.
#'
#' @noRd
#'
#' @importFrom shiny NS tagList  navbarPage div observeEvent reactive tabPanel sidebarLayout sidebarPanel
#' @importFrom shiny selectInput tags verbatimTextOutput absolutePanel req observe
#' @importFrom leaflet leaflet leafletOutput renderLeaflet addProviderTiles setView
#' @importFrom leaflet leafletOptions addPolygons labelOptions highlightOptions
#' @importFrom  leaflet tileOptions leafletProxy addLegend clearControls
#' @importFrom htmltools HTML tags
#' @importFrom glue glue
#' @importFrom dplyr filter
#' @import shiny
#' @importFrom shinyscreenshot screenshot
#'
mod_main_maps_ui <- function(id){
  ns <- NS(id)
  tagList(

tagList(
                    tags$style(type = "text/css", "#main_maps_1-main_map {height: calc(97vh - 100px) !important;}"),
                    leaflet::leafletOutput(ns("main_map")),

                    shiny::verbatimTextOutput(ns("source_main_map")),
                    tags$head(tags$style("#main_maps_1-source_main_map {color:black; font-size:12px; font-style:italic;
                     max-height: 120px; background: #ffe6cc;}")),

                    shiny::absolutePanel(id = "controls_bin", class = "panel panel-default", fixed= TRUE,
                                         draggable = FALSE,  right = 15, top = 65,
                                         width = 250, height = "auto",
                                         style = "background-color: white;
                                                   opacity: 0.85;
                                                   padding: 20px 20px 20px 20px;
                                                   margin: auto;
                                                   border-radius: 5pt;
                                                   box-shadow: 0pt 0pt 6pt 0px rgba(61,59,61,0.48);
                                                   padding-bottom: 2mm;
                                                   padding-top: 1mm;",

                                         shiny::numericInput(ns("bins"),
                                                             label = "Choose number of Bins",
                                                             min = 3,
                                                             max= 10,
                                                             value = 5,
                                                             step=1)
                    ),


                    shiny::absolutePanel(id = "controls", class = "panel panel-default", fixed= TRUE,
                                         draggable = TRUE, bottom = "auto", right = "auto", left = 70, top = 85,
                                         width = 260, height = "auto",
                                         style = "background-color: white;
                                                   opacity: 0.85;
                                                   padding: 20px 20px 20px 20px;
                                                   margin: auto;
                                                   border-radius: 5pt;
                                                   box-shadow: 0pt 0pt 6pt 0px rgba(61,59,61,0.48);
                                                   padding-bottom: 2mm;
                                                   padding-top: 1mm;",
                                         tags$strong(tags$em(h5("Instructions: "))),
                                         tags$strong(tags$em(h6("First, Select the desired domain"))),
                                         tags$strong(tags$em(h6("The indicators available in the second tab will be updated based on the selected domain"))),
                                         tags$strong(tags$em(h6("Finally, select the required year and survey"))),
                                         # tags$strong(tags$em(h6("Note: Some indicators are available for multiple survey rounds for the same year"))),

                                         shiny::selectInput(ns("family"),
                                                            "Select Domain:",
                                                            choices = unique(Pak_Indicators_Data$domain),
                                                            selected = "Household Welfare"
                                                            ),

                                         shiny::selectInput(ns("stat"),
                                                            "Select Indicator: ",
                                                            choices = unique(Pak_Indicators_Data$indicator)),

                                         shiny::selectInput(ns("time"),
                                                            "Select Year:",
                                                            choices = unique(Pak_Indicators_Data$year_1),
                                                            # selected = median(Pak_Indicators_Data$year),
                                                            width = "100%",
                                                            multiple = FALSE,

                                                            ),

                                         downloadButton(ns("mapdata"), "Data", class= "btn-sm"),
                                         actionButton(ns("screenshot"), "Image",class="btn-sm", icon=icon("camera")),
                                         actionButton(ns("help_map"), "Help", icon= icon('question-circle'), class ="btn-sm"),
                                         br(),

                    )
    )
)

}

#' main_maps Server Functions
#'
#' @noRd
mod_main_maps_server <- function(id){
  moduleServer( id, function(input, output, session){
    ns <- session$ns

    #Lealfet static options
    output$main_map <- leaflet::renderLeaflet({
      # message("rendering local map")
      leaflet::leaflet(options = leaflet::leafletOptions(zoomSnap = 0.20, zoomDelta = 0.20)) %>%
        leaflet::addProviderTiles(provider =  "CartoDB.Voyager", group = "CARTO") %>%
        leaflet::setView(lng=69.5, lat = 30, zoom = 5)
    })


    #Main Map
    #selecting domain
    dom_map <- shiny::reactive({
      Pak_Indicators_Data %>%
        dplyr::filter(domain == input$family)
    })
    #
    #
    #Updating indicators based on seelcted domain
    shiny::observeEvent(dom_map(),{
      req(input$family)
      choices <- unique(dom_map()$indicator)
      shiny::updateSelectInput(
        session = getDefaultReactiveDomain(),
        "stat",
        choices = choices)
    })

    #Selected indicator
    ind_selected <- shiny::reactive({
      req(input$family)
      dom_map() %>%
        dplyr::filter(indicator == input$stat)
    })

    #Updating years based on selected indicator
    shiny::observeEvent(ind_selected(), {
      shiny::req(input$stat)
      updated_years_m <- ind_selected() %>%
        dplyr::filter(!is.na(value))    #<<<<<<<<<<
      shiny::updateSelectInput(
        session = getDefaultReactiveDomain(),
        "time",
        choices =sort(unique(updated_years_m$year_1), decreasing = T)
      )
    })

    #making a reactive function for dataset to be used based on User's selection
    map_data <- shiny::reactive({
      Pak_Indicators_Data %>%
        dplyr::filter(
                      domain == input$family,
                      indicator == input$stat,
                      year_1 == input$time)
    })

    #Labelling for the Map
    labels_map <- shiny::reactive({
      paste0(glue::glue("<b>District</b>: { Pak_Shapfiles$district } </br>"),
             glue::glue("<b>Indicator: </b>"), " ",
             glue::glue("{ round(map_data()$value, 2) }"), " ", glue("{map_data()$units}"),
             sep = "") %>%
        lapply(htmltools::HTML)
    })

    #Color Scheme
    pal_new <- reactive({
      req(unique(map_data()$context) %in% c("negative", "positive"))
      if (unique(map_data()$context) == "negative"){
        rev(grDevices::colorRampPalette(colors = c('#2c7bb6', '#abd9e9', '#ffffbf', '#fdae61', '#d7191c'), space = "Lab")(input$bins))
      } else {
        grDevices::colorRampPalette(colors = c('#2c7bb6', '#abd9e9', '#ffffbf', '#fdae61', '#d7191c'), space = "Lab")(input$bins)
      }
    })

    #breaks defined
    breaks <- reactive({
      req(unique(map_data()$context) %in% c("negative", "positive"))
        stats::quantile(map_data()$value, seq(0, 1, 1 / (input$bins)), na.rm = TRUE) %>%
          unique()
    })

    pal <- reactive ({
      leaflet::colorBin(palette = pal_new(),
               bins= breaks(),
               na.color = "grey",
               domain = NULL,
               pretty = F,
               reverse=T)
    })

    # Pal_legend
    pal_leg <- reactive ({
      leaflet::colorBin(palette = pal_new(),
               bins= breaks(),
               na.color = "grey",
               domain = map_data()[,"value"],
               pretty = FALSE,
               reverse=T
      )
    })

    #Dynamic leaflet
    # shiny::observeEvent(input$time,{

       shiny::observe({

      req(map_data())

      leaflet::leafletProxy("main_map",
                            data=Pak_Shapfiles,
                            deferUntilFlush = TRUE) %>%

        leaflet::clearShapes() %>%
        leaflet::addPolygons(label= labels_map(),
                             labelOptions = leaflet::labelOptions(
                               style = list("font-weight"= "normal",
                                            padding= "3px 8px",
                                            "color"= "black"),
                               textsize= "10px",
                               direction = "auto",
                               opacity = 0.9

                             ),
                             fillColor =  ~pal()(map_data()$value),
                             fillOpacity = 1,
                             stroke = TRUE,
                             color= "white",
                             weight = 1,
                             opacity = 0.9,
                             fill = TRUE,
                             dashArray = c(5,5),

                             smoothFactor = 0.8,
                             highlightOptions = leaflet::highlightOptions(weight= 2.5,
                                                                          color = "darkgrey",
                                                                          fillOpacity = 1,
                                                                          opacity= 1,
                                                                          bringToFront = TRUE),
                             group = "Polygons")


      leaflet::leafletProxy("main_map", data= map_data()) %>%
        leaflet::clearControls() %>%
        leaflet::addLegend("bottomright",
                           pal= pal_leg(),
                           values= map_data()$value,
                          title =
                          if(unique(map_data()$units)!=""){
                          paste0("Indicator", " ","(", unique(map_data()$units), ")")
                           }else{
                           "Indicator"
                           },
                           opacity= 1,
                           labFormat = leaflet::labelFormat(
                             between = " : ",
                             digits = 2))

    })

    # Message on updation of the MAPS
    shiny::observe({
      req(input$family)
      req(input$stat)
      req(input$time)
      shiny::showNotification("MAP is being rendered based on the selection",
                              type="message",
                              duration = 3)
    })


    #Source of the slected indicator
    output$source_main_map <- shiny::renderText({
      paste(" Source: ", glue("{ unique(map_data()$source) }"),
            "\n",
            "Definition: ", glue("{ unique(map_data()$definition) }"))
    })

    #Screenshot
    shiny::observeEvent(input$screenshot,{
      shinyscreenshot::screenshot(filename = glue::glue("{ input$stat }", "_", { "input$time" }),
                                  id = "main_map", scale = 0.90, timer = 1)
    })

    #Download data underlying the shown map
    output$mapdata <- shiny::downloadHandler(
      filename = function(){
        paste(glue::glue("{ input$stat }"), "_", glue::glue("{ input$time }"), ".csv")
      },
      content = function(file){
        write.csv(map_data() %>% dplyr::select(-context, -positive, -negative, district1), file)
      }
    )

#Main Interactive Maps
    ##############################################.
    #### Modal  ----
    ###############################################.
    shiny::observeEvent(input$help_map, {
      shiny::showModal(modalDialog(
        title = "How to use these maps",
        p("These maps give district (admin2) level estimates of various socioeconomic indicators over the selected filters"),
        p("All indicators are rounded to 2 decimal points"),
        # p("All Natural Hazards Indicators are rounded to 3 decimal points"),
        p("Expect the color mapping to reverse with the context of  the selected indicators - e.g. Poverty (High) = Red whereas; Access to improved toilet facilities (High) = Blue"),
        size = "m", easyClose = TRUE, fade=FALSE,footer = modalButton("Close (Esc)")))
    })


  })



}

